central/alert/datastore/internal/search/searcher_impl.go

```diff
@@ -41,22 +45,32 @@ func (ds *searcherImpl) SearchListAlerts(ctx context.Context, q *v1.Query, exclu
 	if excludeResolved {
 		q = applyDefaultState(q)
 	}
-	alerts, err := ds.storage.GetByQuery(ctx, q)
+	listAlerts := make([]*storage.ListAlert, 0, paginated.GetLimit(q.GetPagination().GetLimit(), whenUnlimited))
+	err := ds.storage.GetByQueryFn(ctx, q, func(alert *storage.Alert) error {
+		listAlerts = append(listAlerts, convert.AlertToListAlert(alert))
+		return nil
+	})
 	if err != nil {
 		return nil, err
 	}
-	listAlerts := make([]*storage.ListAlert, 0, len(alerts))
-	for _, alert := range alerts {
-		listAlerts = append(listAlerts, convert.AlertToListAlert(alert))
-	}
 	return listAlerts, nil
 
 }
```

Comment:
nit: do we really need to convert `storage.Alert` to `storage.ListAlert` in the code that calls this function (possibly indirectly: all caller sites of the call stack)? If it is just for the convenience of the reduced interface, maybe using the larger object would be enough.
I could think that the type conversion would be necessary in the service layer just before returning the data to the requester.

Reply:
TBH, I think that is a larger issue of separating storage from API.  IMO, out of scope of this PR.

Reply:
I do agree it is out of the scope of the current PR. I was more thinking out loud on possible improvements here.